pool/pool.go

```diff
@@ -585,7 +576,130 @@ func (p *pool) removeSessionTokenAfterThreshold(cfg *callConfig) error {
 	return nil
 }
 
-func (p *pool) PutObject(ctx context.Context, params *client.PutObjectParams, opts ...CallOption) (*oid.ID, error) {
+type callContext struct {
+	// base context for RPC
+	context.Context
+
+	client Client
+
+	// client endpoint
+	endpoint string
+
+	// request signer
+	key *ecdsa.PrivateKey
+
+	// flag to open default session if session token is missing
+	sessionDefault bool
+	sessionTarget  func(session.Token)
+}
+
+func (p *pool) initCallContext(ctx *callContext, cfg *callConfig) error {
+	cp, err := p.connection()
+	if err != nil {
+		return err
+	}
+
+	ctx.key = cfg.key
+	if ctx.key == nil {
+		// use pool key if caller didn't specify its own
+		ctx.key = p.key
+	}
+
+	ctx.endpoint = cp.address
+	ctx.client = cp.client
+
+	if ctx.sessionTarget == nil && cfg.stoken != nil {
+		ctx.sessionTarget(*cfg.stoken)
```

Comment:
It seems it will panic because of condition `ctx.sessionTarget == nil`

Reply:
Definitely, fixed.